src/stepFunctions/commands/publishStateMachine.ts

```diff
@@ -30,6 +30,11 @@ export async function publishStateMachine(awsContext: AwsContext, outputChannel:
         throw new Error('Could not get active text editor for state machine definition')
     }
 
+    if (textDocument.languageId === 'yasl') {
+        logger.error('Cannot publish state machine from yasl (YAML Amazon States Language) file')
+        throw new Error('Cannot publish state machine from yasl (YAML Amazon States Language) file')
```

Comment:
Is the thrown error surfaced to the user? If so, localize (don't localize the `logger.error` string though).

Nit: there's no such thing as a `.yasl` extension. Change wording to something like `'Cannot publish state machine from Amazon States Language YAML file'`